src/main/kotlin/pl/droidsonroids/gradle/jacoco/testkit/JaCoCoTestKitPlugin.kt

```diff
@@ -2,17 +2,43 @@ package pl.droidsonroids.gradle.jacoco.testkit
 
 import org.gradle.api.Plugin
 import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.TaskProvider
 import org.gradle.testing.jacoco.plugins.JacocoPlugin.DEFAULT_JACOCO_VERSION
 import org.gradle.testing.jacoco.plugins.JacocoPluginExtension
-import pl.droidsonroids.gradle.jacoco.testkit.Configurations.currentTestRuntime
+import org.gradle.testing.jacoco.plugins.JacocoTaskExtension
 import pl.droidsonroids.gradle.jacoco.testkit.Configurations.jacocoRuntime
-import pl.droidsonroids.gradle.jacoco.testkit.Tasks.generateJacocoTestKitProperties
-import pl.droidsonroids.gradle.jacoco.testkit.Tasks.test
+
+open class JaCoCoTestKit(private val project: Project) {
```

Comment:
I would call this class `JacocoTestKitExtension` and extract it to a separate file.